addons/mail/static/src/core/common/attachment_model.js

```diff
@@ -28,6 +30,19 @@ export class Attachment extends FileModelMixin(Record) {
     /** @type {string} */
     ownership_token;
     create_date = fields.Datetime();
+    has_thumbnail = fields.Attr(undefined, {
+        onUpdate() {
+            if (this.isPdf && !this.has_thumbnail) {
+                this.setPdfThumbnail();
+            }
+        },
+    });
+
+    get thumbnailUrl() {
+        return imageUrl("ir.attachment", this.id, "thumbnail", {
+            unique: this.checksum,
+        });
+    }
```

Comment:
Attachment data should contain a thumbnail limited access token to allow guests to see them too.